photon-ml/src/main/scala/com/linkedin/photon/ml/algorithm/CoordinateDescent.scala

```diff
@@ -222,33 +225,36 @@ class CoordinateDescent(
           validatingScoresContainerOption = Some(updatedValidatingScoresContainer)
           val fullScore = updatedValidatingScoresContainer.values.reduce(_ + _)
 
-          val (firstEvaluator, currentEvaluation) = evaluators.map { evaluator =>
+          val currentEvaluation = evaluators.map { evaluator =>
             val evaluationTimer = Timer.start()
             val evaluation = evaluator.evaluate(fullScore.scores)
             logger.debug(s"Finished score evaluation, time elapsed: ${evaluationTimer.stop().durationSeconds} (s).")
             logger.info(s"Evaluation metric computed with ${evaluator.getEvaluatorName} after updating " +
               s"coordinateId $coordinateId at iteration $iteration is $evaluation")
 
             (evaluator, evaluation)
-          }.head // we use only the first evaluator to select the "best" model
+          }
 
           logger.debug(s"Finished validating model, time elapsed: ${validationTimer.stop().durationSeconds} (s).")
 
-          (firstEvaluator, currentEvaluation)
+          currentEvaluation
         } // validatingDataAndEvaluatorsOption.map
 
         logger.info(s"Updating coordinate $coordinateId finished, time elapsed: " +
             s"${coordinateTimer.stop().durationSeconds} (s)\n")
 
-        currentEvaluation // the first evaluator of the last coordinate evaluates the full model
+        currentEvaluation
```

Comment:
I think I removed this because it's not quite right now -- this will contain all the evaluators, not just the first one.